src/Word/UnderscoreToStudlyCase.php

```diff
@@ -4,50 +4,41 @@
 
 namespace Laminas\Filter\Word;
 
-use Laminas\Stdlib\StringUtils;
+use Laminas\Filter\FilterInterface;
+use Laminas\Filter\ScalarOrArrayFilterCallback;
 
-use function array_map;
 use function is_array;
 use function is_scalar;
 use function mb_strlen;
 use function mb_strtolower;
 use function mb_substr;
 
-/**
- * @psalm-type Options = array{
- *     separator?: string,
- *     ...
- * }
- * @template TOptions of Options
- * @template-extends UnderscoreToCamelCase<TOptions>
- */
-final class UnderscoreToStudlyCase extends UnderscoreToCamelCase
+/** @implements FilterInterface<string|array<array-key, string|mixed>> */
+final class UnderscoreToStudlyCase implements FilterInterface
 {
-    /**
-     * Defined by Laminas\Filter\Filter
-     *
-     * @psalm-return ($value is scalar ? string : $value is array ? array : mixed)
-     */
     public function filter(mixed $value): mixed
     {
         if (! is_scalar($value) && ! is_array($value)) {
             return $value;
         }
 
         /** @var string|array $value */
-        $value          = parent::filter($value);
-        $lowerCaseFirst = 'lcfirst';
+        $value = (new SeparatorToCamelCase('_'))($value);
```

Comment:
Nit picky, but could you call `->filter($value)` here please?

Reply:
No problem. I've amended that